elasticsearch/probe/readiness.sh

```diff
@@ -19,20 +19,45 @@
 # TODO: try re-use code from ./run.sh
 ES_REST_BASEURL=https://localhost:9200
 EXPECTED_RESPONSE_CODE=200
+EXPECTED_SG_DOCUMENT_COUNT=5
 secret_dir=/etc/elasticsearch/secret
 max_time=${READINESS_PROBE_TIMEOUT:-30}
 
-response_code=$(curl -s -X HEAD \
-    --cacert $secret_dir/admin-ca \
-    --cert $secret_dir/admin-cert \
-    --key  $secret_dir/admin-key \
-    --max-time $max_time \
-    -w '%{response_code}' \
-    "${ES_REST_BASEURL}/")
 
-if [ ${response_code} == ${EXPECTED_RESPONSE_CODE} ]; then
-  exit 0
-else
-  echo "Elasticsearch node is not ready to accept HTTP requests yet [response code: ${response_code}]"
+function check_if_ready() {
+  path="$1"
+  err_msg="$2"
+  response_code=$(curl -s --head \
+      --cacert $secret_dir/admin-ca \
+      --cert $secret_dir/admin-cert \
+      --key  $secret_dir/admin-key \
+      --max-time $max_time \
+      -o /dev/null \
+      -w '%{response_code}' \
+      "${ES_REST_BASEURL}${path}")
+
+  if [ "${response_code}" != ${EXPECTED_RESPONSE_CODE} ]; then
+    echo "${err_msg} [response code: ${response_code}]"
+    exit 1
+  fi
+}
+
+check_if_ready "/" "Elasticsearch node is not ready to accept HTTP requests yet"
+check_if_ready "/.searchguard.$DC_NAME" "Searchguard index '.searchguard.$DC_NAME' is missing in ES cluster"
+sg_doc_count=$(curl -s --get \
+  --cacert $secret_dir/admin-ca \
+  --cert $secret_dir/admin-cert \
+  --key  $secret_dir/admin-key \
+  --max-time $max_time \
+  "${ES_REST_BASEURL}/.searchguard.$DC_NAME/_count" \
+  | python -c "import sys, json; print json.load(sys.stdin)['count']")
```

Comment:
`_count` is not supported in ES 5 afaik.  Instead, do a `_search?size=0` and look at the `total` field.

Reply:
appears to be supported: https://www.elastic.co/guide/en/elasticsearch/reference/5.6/search-count.html

Reply:
It seems `_count` [has been removed](https://github.com/elastic/elasticsearch/pull/14166) for Java API only in the end.

Reply:
some information I was able to find on this topic is https://github.com/elastic/elasticsearch/pull/9296. Given that internally, they should be equivalent but the Java API has removed `count` already, I will go with what Rich suggested and make a fix